operators/nvidia_video_codec/nv_video_decoder/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.24)
+project(nv_video_decoder)
+
+find_package(holoscan 3.0 REQUIRED CONFIG
+             PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
+
+# Detect system architecture
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64|ARM64")
+    set(LIB_ARCH_DIR "aarch64-linux-gnu")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|AMD64")
+    set(LIB_ARCH_DIR "x86_64-linux-gnu")
+else()
+    message(FATAL_ERROR "Unsupported architecture: ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+
+find_library(NVCUVID_LIBRARY nvcuvid
+    NAMES libnvcuvid.so.1 libnvcuvid
+    HINTS ${CUDA_TOOLKIT_ROOT_DIR}/lib64
+    PATHS "/usr/lib/${LIB_ARCH_DIR}/")
```

Comment:
Does the Video Codec SDK come with a CMake config file to enable `find_package(nvcuvid)` or similar?

If not we might explore adding a simple `Findnvcuvid.cmake` or similar with the search strategy here for others to reuse, and include an imported target definition for each Video Codec SDK library to handle includes + lib paths.

Reply:
Thanks! I have refactored the relevant code to `cmake/nvidia_video_codec.cmake`